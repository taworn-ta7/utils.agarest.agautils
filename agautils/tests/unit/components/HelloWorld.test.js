import testHelpers from '@@/unit/test-helpers.js';

import HelloWorld from '@/components/HelloWorld.vue';

describe('HelloWorld.vue', () => {
  const message = 'Test';

  const setupWrapper = async () => {
    const requiredProps = { message };
    const options = {
      props: requiredProps
    };
    const wrapper = await testHelpers.shallowMount(HelloWorld, options);
    return wrapper;
  };

  describe('Desktop', () => {
    test('Render default contents', async () => {
      const wrapper = await setupWrapper();

      expect(wrapper)
        .toMatchSnapshot();
    });

    test('Message prop displays correctly', async () => {
      const wrapper = await setupWrapper();

      expect(wrapper.find('[data-test="message"]').text())
        .toEqual(message);
    });

    test('Open dev tools', async () => {
      const wrapper = await setupWrapper();

      const button = wrapper.find('[data-test="devToolsButton"]');
      await button.trigger('click');

      expect(window.nw.Window.get().showDevTools)
        .toHaveBeenCalled();
    });

    test('App opens in default browser', async () => {
      const wrapper = await setupWrapper();

      const button = wrapper.find('[data-test="openAppInBrowserButton"]');
      await button.trigger('click');

      expect(window.nw.Shell.openExternal)
        .toHaveBeenCalledWith('http://localhost:3000/');
    });
  });

  describe('Web', () => {
    beforeEach(() => {
      window.webSetup();
    });

    test('Render default contents', async () => {
      const wrapper = await setupWrapper();

      expect(wrapper)
        .toMatchSnapshot();
    });

    test('Message prop displays correctly', async () => {
      const wrapper = await setupWrapper();

      expect(wrapper.find('[data-test="message"]').text())
        .toEqual(message);
    });
  });
});
