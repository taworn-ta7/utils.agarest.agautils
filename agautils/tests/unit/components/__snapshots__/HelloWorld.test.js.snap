// Vitest Snapshot v1

exports[`HelloWorld.vue > Desktop > Render default contents 1`] = `
<div class="greetings">
  <h1 class="green">Test</h1>
  <h3> This boilerplate uses <external-link-stub name="NW.js" url="nwjs.io"></external-link-stub> + <external-link-stub name="Vite" url="vitejs.dev"></external-link-stub> + <external-link-stub name="Vue 3" url="vuejs.org"></external-link-stub> + <external-link-stub name="Pinia" url="pinia.vuejs.org"></external-link-stub> + the <external-link-stub name="superior Options API" url="docs.google.com/presentation/d/1fsDhn_MptD6c-_ALgurQtvaRFkfbfukbbFGfEfckzvk/edit?usp=sharing"></external-link-stub>. </h3>
  <div class="center mt-1">
    <p class="container">
      <button> Open Vue-DevTools </button>
    </p>
    <p class="container">
      <button> Open App in default browser </button>
    </p>
  </div>
</div>
`;

exports[`HelloWorld.vue > Web > Render default contents 1`] = `
<div class="greetings">
  <h1 class="green">Test</h1>
  <h3> This boilerplate uses <external-link-stub name="NW.js" url="nwjs.io"></external-link-stub> + <external-link-stub name="Vite" url="vitejs.dev"></external-link-stub> + <external-link-stub name="Vue 3" url="vuejs.org"></external-link-stub> + <external-link-stub name="Pinia" url="pinia.vuejs.org"></external-link-stub> + the <external-link-stub name="superior Options API" url="docs.google.com/presentation/d/1fsDhn_MptD6c-_ALgurQtvaRFkfbfukbbFGfEfckzvk/edit?usp=sharing"></external-link-stub>. </h3>
  <p> Notice that the <strong>same app conditionally renders</strong> different content in a normal browser than <strong>when ran as a desktop app</strong>. </p>
</div>
`;
